version: '3'

services:

  redis:
    image: redis
    ports:
      - '6379:6379'

  db:
    image: postgres
    environment:
      POSTGRES_PASSWORD: ${DATABASE_PASSWORD}
      POSTGRES_USER: ${DATABASE_USER}
      PGDATA: /var/lib/postgresql/data/pgdata
      POSTGRES_DB: ${DATABASE_NAME}
    volumes:
      - pg_data:/var/lib/postgresql/data/pgdata
    ports:
      - '5432:5432'
    healthcheck:
      test: [ 'CMD-SHELL', 'pg_isready -U ${DATABASE_USER}' ]
      interval: 10s
      timeout: 5s
      retries: 3

  app:
    build: .
    tty: true
    command: python manage.py migrate && python manage.py runserver 0.0.0.0:8000
    volumes:
      - .:/code
    ports:
      - '8000:8000'
    depends_on:
      db:
        condition: service_healthy

  celery:
    build: .
    environment:
      CELERY_BROKER_URL: $(CELERY_BROKER_URL)
      CELERY_RESULT_BACKEND: $(CELERY_RESULT_BACKEND)
    tty: true
    command: celery -A config worker -l INFO
    depends_on:
      - redis
      - app

  celery_beat:
    build: .
    tty: true
    command: celery -A config beat -l INFO -s django
    depends_on:
      - redis
      - app

volumes:
  pg_data:
